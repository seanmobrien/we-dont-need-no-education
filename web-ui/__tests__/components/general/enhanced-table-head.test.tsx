import React from 'react';
import { render } from '/__tests__/test-utils';
import EnhancedTableHead from '/components/general/enhanced-table-head';

const mockHeadCells = [
  { id: 'name', label: 'Name', numeric: false, disablePadding: false },
  { id: 'email', label: 'Email', numeric: false, disablePadding: false },
  { id: 'count', label: 'Count', numeric: true, disablePadding: false },
];

describe('EnhancedTableHead', () => {
  const defaultProps = {
    headCells: mockHeadCells,
    numSelected: 0,
    rowCount: 10,
  };

  it('renders table head snapshot', () => {
    const { container } = render(
      <table>
        <EnhancedTableHead {...defaultProps} />
      </table>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with some selected items snapshot', () => {
    const { container } = render(
      <table>
        <EnhancedTableHead {...defaultProps} numSelected={2} />
      </table>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with all items selected snapshot', () => {
    const { container } = render(
      <table>
        <EnhancedTableHead {...defaultProps} numSelected={10} />
      </table>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with select all callback snapshot', () => {
    const { container } = render(
      <table>
        <EnhancedTableHead {...defaultProps} onSelectAllClick={jest.fn()} />
      </table>,
    );
    expect(container).toMatchSnapshot();
  });
});
