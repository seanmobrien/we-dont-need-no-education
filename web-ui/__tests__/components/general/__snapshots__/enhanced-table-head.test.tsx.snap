// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`EnhancedTableHead renders table head snapshot 1`] = `
<div>
  <table>
    <thead
      class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1t6vsw9-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-xncyw9-MuiTableCell-root"
          scope="col"
          vertical-align="center"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-xncyw9-MuiTableCell-root"
          scope="col"
          vertical-align="center"
        >
          Email
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1o9hn7v-MuiTableCell-root"
          scope="col"
          vertical-align="center"
        >
          Count
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`EnhancedTableHead renders with all items selected snapshot 1`] = `
<div>
  <table>
    <thead
      class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1t6vsw9-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-xncyw9-MuiTableCell-root"
          scope="col"
          vertical-align="center"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-xncyw9-MuiTableCell-root"
          scope="col"
          vertical-align="center"
        >
          Email
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1o9hn7v-MuiTableCell-root"
          scope="col"
          vertical-align="center"
        >
          Count
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`EnhancedTableHead renders with select all callback snapshot 1`] = `
<div>
  <table>
    <thead
      class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1t6vsw9-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-j0ceml-MuiTableCell-root"
          scope="col"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-1poq2nf-MuiButtonBase-root-MuiSwitchBase-root-MuiCheckbox-root"
          >
            <input
              aria-label="select all emails"
              class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
              data-indeterminate="false"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1fm565k-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-xncyw9-MuiTableCell-root"
          scope="col"
          vertical-align="center"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-xncyw9-MuiTableCell-root"
          scope="col"
          vertical-align="center"
        >
          Email
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1o9hn7v-MuiTableCell-root"
          scope="col"
          vertical-align="center"
        >
          Count
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`EnhancedTableHead renders with some selected items snapshot 1`] = `
<div>
  <table>
    <thead
      class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1t6vsw9-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-xncyw9-MuiTableCell-root"
          scope="col"
          vertical-align="center"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-xncyw9-MuiTableCell-root"
          scope="col"
          vertical-align="center"
        >
          Email
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1o9hn7v-MuiTableCell-root"
          scope="col"
          vertical-align="center"
        >
          Count
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
